whisperv6/whisper.go

```diff
@@ -800,8 +803,20 @@ func (whisper *Whisper) runMessageLoop(p *Peer, rw p2p.MsgReadWriter) error {
 			log.Warn("unxepected status message received", "peer", p.peer.ID())
 		case messagesCode:
 			// decode the contained envelopes
+			data, err := ioutil.ReadAll(packet.Payload)
+			if err != nil {
+				log.Warn("failed to read envelopes data", "peer", p.peer.ID(), "error", err)
+				return errors.New("invalid enveloopes")
+			}
+			if !whisper.noConfirmations {
+				batchHash := crypto.Keccak256Hash(data)
```

Comment:
I would encapsulate logic to create batch hashes so that it's easy to refactor if needed.